spec/api/sul_bib/authorship_api_spec.rb

```diff
@@ -44,44 +43,24 @@
   # method path, json, headers
   # e.g.
   # let(:http_request) do
-  #   post '/authorship', json_data, headers
+  #   post '/authorship', request_data.to_json, headers
   # end
 
-  shared_examples 'it is successful' do
-    it 'has response code < 300' do
-      http_request # defined in calling context
-      expect(response.status).to be < 300
-    end
-  end
-
-  shared_examples 'it creates a new contribution' do
-    it 'it increases number of contribution records by one' do
-      expect do
-        http_request # defined in calling context
-      end.to change(Contribution, :count).by(1)
-    end
-  end
-
   shared_examples 'it creates new contributions and publications' do
     context 'with no other contributions' do
       let(:request_data) { valid_data_for_post.merge(author_hash) }
-      it_behaves_like 'it is successful'
-      it_behaves_like 'it creates a new contribution'
+      it 'successfully creates one new contribution' do
+        expect { http_request }.to change(Contribution, :count).by(1)
+        expect(response.status).to eq(201)
+      end
     end
 
     context 'with prior contributions' do
-      let(:request_data) { new_authorship_for_pub_with_contributions.merge(author_hash) }
-      it_behaves_like 'it is successful'
-      it_behaves_like 'it creates a new contribution'
-      it 'creates a new authorship record without overwriting existing authorship records' do
-        http_request
-        count = contribution_count + 1
-        expect(publication_with_contributions.contributions.reload.size).to eq(count)
-      end
-      it 'increases number of contribution records for specified publication by one' do
-        expect do
-          http_request
-        end.to change(publication_with_contributions.contributions, :count).by(1)
+      let(:request_data) { base_data.merge(sul_pub_id: publication_with_contributions.id).merge(author_hash) }
+      it 'successfully increases the publication\'s contribution records by one' do
+        expect { http_request }.to change(publication_with_contributions.contributions, :count).by(1)
+        expect(response).to have_http_status(:success)
+        expect(publication_with_contributions.contributions.count).to eq(contribution_count + 1)
       end
```

Comment:
I don't get how this clarifies stuff any - looks like it could be calling the new shared example and be done with it?  But the shared example now expects a 201 whereas this one uses a different rspec dialect to check for success.  So, if this one is not a 201, then change the shared example to use this rspec dialect and have this use the shared example?  I suppose this is an example of what it means to change some shared-example nesting, but this looks more complicated and less shared.  The whole point of shared examples is to consolidate code, not duplicate it.  Like, lines 61 and 63 in the new branch seem to do the same thing, don't they?

Agreed that using the `expect {}.to change` is a good pattern when we don't care about details.  (I might have coded some of this stuff in my early adjustments to using rspec, it looks like some things I would do differently now.)

Reply:
Not *exactly* the same thing.  The former checks that the increment is 1.  The latter checks the total value.  I agree it might be overkill, but the tests were both there previously.  Want to get rid of one?

Reply:
OK, consolidate where possible.  Maybe there was excess paranoia there before.